transaction/src/action.rs

```diff
@@ -33,6 +35,10 @@ pub use vote::{DelegatorVote, ValidatorVote, ValidatorVoteBody, Vote};
 /// Common behavior between Penumbra actions.
 pub trait IsAction {
     fn balance_commitment(&self) -> balance::Commitment;
+    fn decrypt_with_perspective(
+        &self,
+        txp: &TransactionPerspective,
+    ) -> anyhow::Result<Option<ActionView>>;
```

Comment:
What's the meaning of the `Result<Option<ActionView>>` here? What does an error indicate, vs. an inner `None` value?

Should there be an error at all? In what sense would the method be fallible?

Also, since this produces an `ActionView`, maybe we should call it `view_with_perspective` to be consistent with the metaphor?

Reply:
On the fallibility front, the current behavior of the implementation for `Output`s is to error if the output is not viewable from this perspective, xref https://github.com/penumbra-zone/penumbra/pull/1426/files#r982852861 

But there are normal cases where the output is not viewable, like when someone is viewing an inbound transaction.  In that case, bubbling up an error doesn't seem right.  What's the intended way to handle opaque actions?

Reply:
Yes, this needs taken into account & refactored for the desired behavior for each specific case.